docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -100,23 +83,20 @@ spec:
           secret: 123
 status: 
   vip:
-    ip: 242.0.0.1
+    value: 242.0.0.1
     type: Kuma
   addresses:
-  - address: example.ext.svc.local
-    status: Available
-  - address: httpbin.com
-    status: Available
-  - address: 192.168.0.1
-    status: Available
-  - address: 10.1.1.0/24
+  - hostname: example.ext.svc.local
     status: NotAvailable
+    origin:
+      kind: HostnameGenerator
+      name: k8s-example-service-hostname
     reason: "addresses are overlapping with ext2"
 ```
 * **spec**:
-  * **match**: defines a list of internalVIPs/domains/CIDR/IPs that should be routed through the sidecar
-    * **type**: type of the match, one of `InternalVIP`, `Domain`, `CIDR` and `IP` are available
-      * `InternalVIP`: allocates a VIP for a domain provided in the `value` field
+  * **match**: defines a traffic that should be routed through the sidecar
+    * **type**: type of the match, one of `KumaHostname`, `Domain`, `CIDR` and `IP` are available
+      * `KumaHostname`: allocates a VIP for a domain provided in the `value` field
```

Comment:
I think we changed course and this is not updated yet. It should be something like "matches a hostname provided by HostnameGenerator".